### PR TITLE
feat: allow setting offset without limit

### LINKED ISSUE #891: Setting offset but not limit returns zero rows
Right now we default to limit zero if offset is set, but that's not very useful.

### PR DESCRIPTION
**Pull request by wjones127**
Closes #891.

### CODE REVIEW

File: rust/src/dataset/scanner.rs [hunk 1]
```diff
@@ -216,16 +216,16 @@ impl Scanner {
     }
 
     /// Set limit and offset.
-    pub fn limit(&mut self, limit: i64, offset: Option<i64>) -> Result<&mut Self> {
-        if limit < 0 {
+    pub fn limit(&mut self, limit: Option<i64>, offset: Option<i64>) -> Result<&mut Self> {
+        if limit.unwrap_or(0) < 0 {
```
Comment (eddyxu) [inline rust/src/dataset/scanner.rs:220]: can this be `unwrap_or_default()` ?

File: rust/src/dataset/scanner.rs [hunk 2]
```diff
@@ -769,7 +769,7 @@ mod test {
 
         let dataset = Dataset::open(path).await.unwrap();
         let mut scanner = dataset.scan();
-        scanner.limit(2, Some(19)).unwrap();
+        scanner.limit(Some(2), Some(19)).unwrap();
```
Comment (eddyxu) [inline rust/src/dataset/scanner.rs:772]: what does it mean if we pass `limit(None, Some(19)`?

Comment (wjones127) [reply rust/src/dataset/scanner.rs:772]: Skip the first 19 rows and return everything after.

**Review by eddyxu — APPROVED** [review]
Thanks for the explanation.  LGTM. Pending CI.

**Pull request merged**